modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/support/ArrayValuesSourceAggregatorFactory.java

```diff
@@ -58,17 +59,27 @@ public Aggregator createInternal(SearchContext searchContext,
         if (valuesSources.isEmpty()) {
             return createUnmapped(searchContext, parent, metadata);
         }
-        return doCreateInternal(valuesSources, searchContext, parent, collectsFromSingleBucket, metadata);
+        return doCreateInternal(valuesSources, searchContext, parent, cardinality, metadata);
     }
 
+    /**
+     * Create the {@linkplain Aggregator} for a field that isn't mapped.
```

Comment:
As noted on `doCreateInternal`, it's more accurate to say this is used when we have no values.

Reply:
++